modelator/src/model/language/tla/mod.rs

```diff
@@ -87,7 +92,12 @@ impl Tla {
         // generate a tla test file and config for each test name found
         test_names
             .into_iter()
-            .map(|test_name| Self::generate_test(&test_name, tla_file_suite))
+            .map(|test_name| {
+                Ok(TlaTest {
```

Comment:
write it like

```
Ok(TlaTest {
  file_suite: Self::generate_test(&test_name, tla_file_suite)?,
  name: test_name
})
```

You will avoid unnecessary clone :)

Reply:
Hey, I'm enjoying all the rusty tips you're giving me! W.r.t this one though - I was wondering, this will be a semantic move (with will probably be implemented with a memory copy), and that will surely be optimized out by rustc?

Nonetheless it's obviously a good idea to simplify it.

Reply:
I am not really sure. but we can try looking at the compiled assembly code to confirm :)

Reply:
after a quick check, the `release` binary seems to optimize it,  but `debug` binary doesn't.